### ISSUE TITLE
Embed indexing expression should validate that the embedder id exist

### ISSUE THREAD

**Issue by jobergum** [vespa-engine/vespa#28258]
**Describe the bug**

```
field embedding type tensor<float>(x[384]) {
        indexing: input text | embed foo | attribute | index
        attribute {
            distance-metric: angular
        }
    }
```

Deploys fine, even if the `foo` embedder is not defined in service.xml - this leads to a run-time error during indexing

```
feed: got status 502 ({"pathId":"/document/v1/music/track/docid/TRAPUZT12903CBBB51","id":"id:music:track::TRAPUZT12903CBBB51","message":"[UNKNOWN(252001) @ tcp/vespa-container:19101/chain.indexing]: Processing failed. Error message: java.lang.IllegalStateException: Can't find embedder 'foo'. Valid embedders are something -- See Vespa log for details. "}) for put id:music:track::TRAPUZT12903CBBB51: giving up after 10 attempts
```

**Comment by bjorncs** [comment]
The config model does not know whether a component is an embedder, as an embedder is just a generic component that implements the `Embedder` interface. If referring to an unknown embedder should fail at deploy time, the best approach is to fail startup of the container.

Alternatively we add a validator in the config model that verifies that a component 'foo' exists, without checking whether it's an embedder or not.

**Comment by bratseth** [comment]
Yes, I think we should check during config model construction that the component id we refer to is defined in services.xml.

(The rest, which is more general, should be solved by improving the information flow from the containers back to deployment and in dev/perf wait for startup before declaring deployment a success.)

**Comment by bjorncs** [comment]
[#34181](https://github.com/vespa-engine/vespa/pull/34181) will catch errors such as typos in embedder name or missing embedder component altogether.
